boards/ti/lp_mspm0g3507/docs/index.rst

```diff
@@ -0,0 +1,194 @@
+| SYSTICK   | on-chip    | system clock          |
++-----------+------------+-----------------------+
+| UART      | on-chip    | serial                |
++-----------+------------+-----------------------+
+| GPIO      | on-chip    | gpio                  |
++-----------+------------+-----------------------+
+| PINMUX    | on-chip    | pinctrl               |
++-----------+------------+-----------------------+
+| CLOCKMUX  | on-chip    | clockctl              |
++-----------+------------+-----------------------+
+
+More details about the supported peripherals are available in `MSPM0G3507 TRM`_.
+Other hardware features are not currently supported by the Zephyr kernel.
+
+Building and Flashing
+*********************
+
+Building
+========
+
+Follow the :ref:`getting_started` instructions for Zephyr application development.
+
+For example, to build the blinky application for the MSPM0G3507 LaunchPad:
+
+.. zephyr-app-commands::
+   :zephyr-app: samples/basic/blinky
+   :board: lp_mspm0g3507
+   :goals: build
+
+The resulting ``zephyr.bin`` binary in the build directory can be flashed onto
+MSPM0G3507 LaunchPad using the steps mentioned below.
+
+Flashing
+========
+
+Open OCD is used to program the flash memory on the devices. It may be necessary in earlier versions to use a branch of open OCD onto the device.
+
+Before OpenOCD is public, one can clone `This Repo <https://github.com/nmenon/openocd/tree/mspm0>`_, and then this can be built with
+
+```
+cd <cloned_OPENOCD_dir>
+./bootstrap (when building from the git repository)
+./configure --enable-xds110
+make
+sudo make install
+```
+
+Then after the build, it is possible to flash the device by passing additional arguments to the flash command
+
+```
+west flash --openocd <path to cloned dir>/src/openocd --openocd-search <path to cloned dir>/tcl
+```
+
+
+`UniFlash`_ can also be used to program the flash memory if needed. The flash utility,
+however, requires the use of 8-byte aligned words, so adding alignment to the linker file is necessary.
+
+Debugging
+=========
+
+The flashing method described above does not include symbols. Thus, debugging requires an additional step to load the symbols.
+This section shows how to debug the MSPM0G3507 LaunchPad board using `CCS IDE`_. More information
+on debugging using CCS can be found in `CCS User's Guide`_.
+
+In general, the steps for debugging in CCS are:
+
+   1. Open CCS
+   2. Go to :menuselection:`Window --> Show View --> Target Configruation`
+   3. Import target confguration by right clicking User Defined, selecting Import target configuration and pointing to the lp_mspm0g3507/support/MSPM0G3507.ccxml
+   4. Launch target configuration by right clicking the new MSPM0G3507.ccxml file and clicking Launch target configuration
+   5. Plug in the device and connect to it by going to :menuselection:`Run --> Connect Target`
+   6. Go to :menuselection:`Run --> Load --> Load Symbols and load in the zephyr.elf file loaded`
+   7. Use CCS to debug
```

Comment:
```
west debug --openocd <path to cloned dir>/src/openocd --openocd-search <path to cloned dir>/tcl
```
debugging works fine with openocd as well, add that to doc?. but sometimes it fails with,
```
Error: [mspm0x.cpu] Can not find free FPB Comparator!                                                                                                                                                               
Error: [mspm0x.cpu] can't add breakpoint: resource not available                                                                                                                                                    
[mspm0x.cpu] Can not find free FPB Comparator!                                                                                                                                                                      
[mspm0x.cpu] can't add breakpoint: resource not available                                                                                                                                                           
Error: [mspm0x.cpu] Can not find free FPB Comparator!                                                                                                                                                               
Error: [mspm0x.cpu] can't add breakpoint: resource not available                                                                                                                                                    
[mspm0x.cpu] Can not find free FPB Comparator!                                                                                                                                                                      
[mspm0x.cpu] can't add breakpoint: resource not available 
```
which am not sure.

Reply:
It should be fine. If you could add the reference to the doc for `west debug` it should be enough for now.